src/cargo-fmt/main.rs

```diff
@@ -364,11 +364,16 @@ fn get_targets_root_only(
     targets: &mut BTreeSet<Target>,
 ) -> Result<(), io::Error> {
     let metadata = get_cargo_metadata(manifest_path)?;
+    // `workspace_root_path` is the path to the workspace's root directory
     let workspace_root_path = PathBuf::from(&metadata.workspace_root).canonicalize()?;
     let (in_workspace_root, current_dir_manifest) = if let Some(target_manifest) = manifest_path {
+        // `target_manifest` is the canonicalized path to a `Cargo.toml` file
+        let target_manifest = target_manifest.canonicalize()?;
         (
-            workspace_root_path == target_manifest,
-            target_manifest.canonicalize()?,
+            target_manifest
+                .parent()
+                .is_some_and(|manifest| workspace_root_path == manifest),
```

Comment:
The way this check is currently implemented a user could be in a sub package and run `cargo fmt --manifest-path ../path/to/root/Cargo.toml` and that would find all the targets and format them. I'm not sure if that's something we want to allow, or if we should check that the user's `env::current_dir()` is also in the workspace's root.

Reply:
In fact, a user could be in a sub-package of one workspace and specify `cargo fmt --manifest-path ../../path/to/some/other/projects/Cargo.toml`, and I think that'll format the files in the other project. Again, just want to double check if that's expected behavior.